### PR TITLE
fix Electron support

### PR DESCRIPTION
**Pull request by ndeloof**
this is a proposed fix for #283

`shelljs.exec` is replaced by `execa.sync`

### CODE REVIEW

**Comment by ndeloof** [comment]
I can't find guidance on commit-message format to be used for contribution

**Comment by ndeloof** [comment]
@silasbw forgot this one indeed. And will need to parse cmd-args into a string[]

**Comment by ndeloof** [comment]
Last commit comes with an issue : for `execa.sync` to run the command the `cmd-args` string has to be parsed into individual arguments.

File: src/cloud_auth.ts [hunk 1]
```diff
@@ -53,16 +54,17 @@ export class CloudAuth implements Authenticator {
         if (!config['cmd-path']) {
             throw new Error('Token is expired!');
         }
-        const args = config['cmd-args'];
+        let args: string[] = [];
+        const cmdargs = config['cmd-args'];
+        if (cmdargs) {
+            args = cmdargs.split(' ');
```
Comment (brendandburns) [inline src/cloud_auth.ts]: This is going to break if the string is something like:

`"foo bar" baz`

**Comment by drubin** [comment]
/approve cancel
To address comments

File: src/cloud_auth.ts (hunk 1 above)
Comment (ndeloof) [reply src/cloud_auth.ts]: right, so I checked how the native CLI do implement the same and it doesn't handle this "corner-case": https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/plugin/pkg/client/auth/gcp/gcp.go#L145
So I was assuming this is safe to do the same

Comment (ndeloof) [reply src/cloud_auth.ts]: updated PR to rely on `child_process`, who consumes a plain string command (not individual args) and actualy makes code simpler here.
Alternatively could use string-argv at the cost of an extra dependency.

**Comment by drubin** [comment]
Why did we leave shelljs as a dependancy here as well?

**Comment by ndeloof** [comment]
Still required to implement `which`.
Maybe could be removed with a workaround

**Comment by drubin** [comment]
@ndeloof thanks forgot we did that.

**Pull request merged**